app/scripts/services/tile-proxy.js

```diff
@@ -34,76 +34,49 @@ export let authHeader = null;
  */
 
 /**
+ * Returns a function that batches and throttles calls with a final debounce.
+ *
+ * Takes a function that processes a batch of items and returns a version that accumulates inputs,
+ * executing them at a fixed interval with a final debounce to ensure all items are processed.
+ *
+ * @template T
  * @template {Array<unknown>} Args
  *
- * @param {(ctx: RequestContext, ...args: Args) => void} func
- * @param {number} interval
- * @param {number} finalWait
+ * @param {(items: Array<T>, ...args: Args) => void} func
+ * @param {{ interval: number, finalWait: number }} options
  */
-const throttleAndDebounce = (func, interval, finalWait) => {
+const delayedBatchExecutor = (func, { interval, finalWait }) => {
   /** @type {ReturnType<typeof setTimeout> | undefined} */
   let timeout = undefined;
-  /** @type {Array<WrappedTilesRequest>} */
-  let bundledRequest = [];
-  /** @type {Record<string, number>} */
-  let requestMapper = {};
+  /** @type {Array<T>} */
+  let items = [];
   /** @type {number} */
   let blockedCalls = 0;
 
-  /**
-   * @param {WrappedTilesRequest} request
-   * @returns {void}
-   */
-  const bundleRequests = (request) => {
-    const requestId = requestMapper[request.id];
-
-    if (requestId && bundledRequest[requestId]) {
-      bundledRequest[requestId].ids = bundledRequest[requestId].ids.concat(
-        request.ids,
-      );
-    } else {
-      requestMapper[request.id] = bundledRequest.length;
```

Comment:
This is the main change. This function doesn't need to do this aggregation. It's a separate concern, which we now handle in the batch processor. This allows this function to be generic.